internal/docker/docker_impl.go

```diff
@@ -358,6 +358,69 @@ loop:
 	return err
 }
 
+func (d *dockerV20Container) writeFile(path string, content []byte) error {
+	if d.config.Execution.Mode == config.DockerExecutionModeSession && d.config.Execution.DisableAgent {
+		return message.NewMessage(
+			message.EDockerWriteFileFailed,
+			"Agent needs to be enable in order to write files",
```

Comment:
`"The ContainerSSH guest agent is disabled and session mode is active. Failed to write file %s in the container."`